lib/tiniergltf/tiniergltf.hpp

```diff
@@ -1354,4 +1370,123 @@ struct GlTF {
 	}
 };
 
+// std::span is C++ 20, so we roll our own little struct here.
+template <typename T>
+struct Span {
+	T *ptr;
+	uint32_t len;
+	bool empty() const {
+		return len == 0;
+	}
+	T *end() const {
+		return ptr + len;
+	}
+	template <typename U>
+	Span<U> cast() const {
+		return {(U *) ptr, len};
+	}
+};
+
+static Json::Value readJson(Span<const char> span) {
+	Json::CharReaderBuilder builder;
+	const std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+	Json::Value json;
+	JSONCPP_STRING err;
+	if (!reader->parse(span.ptr, span.end(), &json, &err))
+		throw std::runtime_error("invalid JSON");
```

Comment:
shouldn't the error message be forwarded? currently, you get:

```
ERROR[Main]: Irrlicht: glTF loader: invalid JSON
```

```suggestion
		throw std::runtime_error(std::string("invalid JSON: ").append(err));
```

would give

```
ERROR[Main]: Irrlicht: glTF loader: invalid JSON: * Line 2, Column 1
  Missing '}' or object member name

```

Reply:
sure, that's better